staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_restart.go

```diff
@@ -183,7 +182,8 @@ func (o RestartOptions) RunRestart() error {
 		}
 
 		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
-			allErrs = append(allErrs, fmt.Errorf("failed to create patch for %v: empty patch", info.Name))
+			allErrs = append(allErrs, fmt.Errorf("failed to create patch for %v: if restart has already been triggered within the past second, please wait before attempting to trigger another", info.Name))
+			continue
```

Comment:
This `continue` fixes unnecessary and misleading `restarted` message like in the issue;

```
10:22:35  deployment.apps/vcita-automation restarted
10:22:35  deployment.apps/vcita-automation-resque restarted
10:22:35  deployment.apps/vcita-automation-resque-scheduler restarted
10:22:35  failed to create patch for vcita-automation: empty patch
10:22:35  failed to create patch for vcita-automation-resque: empty patch
10:22:35  failed to create patch for vcita-automation-resque-scheduler: empty patch
```

Reply:
Message looks good to me. Could you please add a test for this empty patch and include an assertion against the regression you're describing above (i.e. we don't print "restarted" when we didn't)?

Reply:
I can add unit test for empty patch scenario. But since this depends on timing, I hope it will not create any flakiness

Reply:
It looks like `RestartOptions.Restarter` might be able to help with that? You could supply a restarter that never returns a change?

Reply:
@KnVerey I added unit test to cover this patch. Could you PTAL?, thanks.